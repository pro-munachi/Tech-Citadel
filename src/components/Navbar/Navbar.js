import { Box, useColorModeValue, useColorMode, Flex } from "@chakra-ui/react";
import React from "react";

const Navbar = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  return (
    <Flex
      h="80px"
      bg={useColorModeValue("lightMode.mainBgColor", "darkMode.mainBgColor")}
      justifyContent="space-between"
      alignItems="center"
    >
      <Box>School</Box>

      <Box onClick={() => toggleColorMode()}>
        {colorMode === "light" ? "Darkmode" : "Lightmode"}
      </Box>
    </Flex>
  );
};

export default Navbar;
