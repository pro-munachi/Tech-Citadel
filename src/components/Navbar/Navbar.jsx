import {
  Box,
  useColorModeValue,
  useColorMode,
  Flex,
  Button,
} from "@chakra-ui/react";
import React from "react";
import { Link } from "react-router-dom";

const Navbar = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  return (
    <Flex
      h="80px"
      bg={useColorModeValue("lightMode.navBgColor", "darkMode.navBgColor")}
      justifyContent="space-between"
      alignItems="center"
      color="white"
    >
      <Box
        ml="15px"
        fontSize={["20px", "20px", "28px", "28px"]}
        data-testid="logo"
      >
        <Link to="/">Citadel</Link>
      </Box>

      <Box onClick={() => toggleColorMode()} mr="15px" data-testid="toggle">
        <Button>{colorMode === "light" ? "Darkmode" : "Lightmode"}</Button>
      </Box>
    </Flex>
  );
};

export default Navbar;
