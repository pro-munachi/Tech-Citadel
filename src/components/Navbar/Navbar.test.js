import { render, screen } from "@testing-library/react";
import Navbar from "./Navbar";
import { BrowserRouter } from "react-router-dom";
import { ChakraProvider, theme } from "@chakra-ui/react";
import store from "../../state/store";
import { Provider } from "react-redux";

const Wrapper = ({ children }) => {
  <ChakraProvider theme={theme}>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </ChakraProvider>;
};

test("Test Navbar", async () => {
  const { asFragment } = render(
    <Wrapper>
      <Navbar />
    </Wrapper>
  );

  expect(asFragment()).toMatchSnapshot();
});
