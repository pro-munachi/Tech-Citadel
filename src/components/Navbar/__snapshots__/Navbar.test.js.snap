// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar 1`] = `<DocumentFragment />`;
